### PR TITLE
Update quick start guide && remove legacy bash files

### PR DESCRIPTION
**Pull request by flyinghorse0510**
Update quick start guide to use Go version scripts
Remove legacy bash files under `scripts`

### CODE REVIEW

**Comment by leokondrashov** [comment]
As I understand this PR depends on #835, right? Since scripts are completely removed.

And to add, this PR closes #821.

As for the review, please fix linter's errors.

**Comment by flyinghorse0510** [comment]
> As I understand this PR depends on #835, right? Since scripts are completely removed.
> 
> And to add, this PR closes #821.
> 
> As for the review, please fix linter's errors.

Actually it doesn't depend on #835 because in this PR only some useless bash scripts without dependency issues are removed.
But to remove other bash scripts further, #835 need to be done firstly.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,11 @@
 ### Added
 - Added support for [NVIDIA GPU](https://docs.nvidia.com/datacenter/cloud-native/kubernetes/install-k8s.html) in stock-only setup, with [setup script](./scripts/gpu/setup_nvidia_gpu.sh) and [example](./configs/gpu/gpu-function.yaml) Knative deployment 
 - Bumped gVisor to 20230911.0.
+- Refactored bash setup scripts with [golang](./scripts/setup.go) which provide a unified entry for easily setting up vHive.
 ### Changed
 - Changed [system setup script](./scripts/setup_system.sh). NVIDIA helm is now one of the vHive dependencies.
+- Remove some useless bash scripts under [scripts](./scripts/)
```
Comment (ustiugov) [inline CHANGELOG.md:11]:
```suggestion
- Remove some legacy bash scripts under [scripts](./scripts/)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,8 +5,11 @@
 ### Added
 - Added support for [NVIDIA GPU](https://docs.nvidia.com/datacenter/cloud-native/kubernetes/install-k8s.html) in stock-only setup, with [setup script](./scripts/gpu/setup_nvidia_gpu.sh) and [example](./configs/gpu/gpu-function.yaml) Knative deployment 
 - Bumped gVisor to 20230911.0.
+- Refactored bash setup scripts with [golang](./scripts/setup.go) which provide a unified entry for easily setting up vHive.
```
Comment (ustiugov) [inline CHANGELOG.md:8]: move to changed and squash with the removal

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,8 +5,11 @@
...
 ### Changed
 - Changed [system setup script](./scripts/setup_system.sh). NVIDIA helm is now one of the vHive dependencies.
+- Remove some useless bash scripts under [scripts](./scripts/)
+- Update [quick start guide](./docs/quickstart_guide.md) to use refactored Go version setup scripts
```
Comment (ustiugov) [inline CHANGELOG.md:12]: too minor, remove. should 1 line in total

File: docs/quickstart_guide.md [hunk 4]
```diff
@@ -5,29 +5,34 @@ See [here][github-toc] to learn where to find table of contents.
...
-    2. [Clean Up](#2-clean-up)
-    3. [Using a Script](#3-using-a-script)
-4. [Deploying and Invoking Functions in vHive](#iv-deploying-and-invoking-functions-in-vhive)
-    1. [Deploy Functions](#1-deploy-functions)
-    2. [Invoke Functions](#2-invoke-functions)
-    3. [Delete Deployed Functions](#3-delete-deployed-functions)
-5. [Deploying eStargz-based Functions](#v-deploying-estargz-based-functions)
-    1. [Deploy and Invoke Functions](#1-deploy-and-invoke-functions)
-    2. [Delete Deployed Function](#2-delete-deployed-function)
+- [vHive Quickstart](#vhive-quickstart)
```
Comment (ustiugov) [inline docs/quickstart_guide.md:8]: let's just remove this Contents as we have this button on the right.

File: docs/quickstart_guide.md [hunk 5]
```diff
@@ -67,21 +73,52 @@ We tested the following instructions by setting up a **2-node** cluster on Cloud
 
 SSD-equipped nodes are highly recommended. Full list of CloudLab nodes can be found [here][cloudlab-hw].
 
+### 4. Go Installation
```
Comment (ustiugov) [inline docs/quickstart_guide.md:76]: this subsection is an unnecessary complication. we should just refer people to install the supported from the official instructions. Also, I don't think we support 1.20... can you check the dev guide where all verreions are listed and the CI to see the exact version?

File: docs/quickstart_guide.md [hunk 6]
```diff
@@ -222,7 +264,7 @@ SSD-equipped nodes are highly recommended. Full list of CloudLab nodes can be fo
     > ```
 
 ### 5. Finalise Master Node
-**On the master node**, execute the following instructions below **as a non-root user with sudo rights** using **bash**:
+**On the master node**, execute the following instructions below **as a non-root user with sudo access with sudo rights** using **bash**:
```
Comment (ustiugov) [inline docs/quickstart_guide.md:267]: redundant change

File: docs/quickstart_guide.md [hunk 7]
```diff
@@ -239,16 +281,16 @@ In essence, you will execute the same commands for master and worker setups but
 
 5 seconds delay has been added between the commands to ensure that components have enough time to initialize.
 
-Execute the following below **as a non-root user with sudo rights** using **bash**:
+Execute the following below **as a non-root user with sudo access** using **bash**:
```
Comment (ustiugov) [inline docs/quickstart_guide.md:284]: redundant change

**Review by ustiugov — CHANGES_REQUESTED** [review]

**Comment by leokondrashov** [comment]
There is a broken link in the CHANGELOG.md:82. I propose changing it to the quickstart guide section of creating one node cluster. @ustiugov or should we just remove the link from the file?

File: docs/quickstart_guide.md [hunk 8]
```diff
@@ -67,21 +43,41 @@ We tested the following instructions by setting up a **2-node** cluster on Cloud
...
+### 4. Go Installation
+
+If you intend to build the setup scripts from source, you can refer to the [Go Installation Guide](https://go.dev/doc/install) for installing Go on your system.
+
+- Confirm the installation:
+   ```bash
+   go version
+   ```
+
+**We highly recommend you install Go with version `1.19`**
```
Comment (leokondrashov) [inline docs/quickstart_guide.md:55]: Won't it make go being installed twice: here and during setup_node script? And that actually makes little sense to install go from go tool.

Comment (ustiugov) [reply docs/quickstart_guide.md:55]: I agree with that. Leonid, could you fix up and go ahead with merging this PR? sync up with Ruiqi about the runners

Comment (ustiugov) [reply docs/quickstart_guide.md:55]: @leokondrashov

**Pull request merged**